src/cdomains/exp.ml

```diff
@@ -59,6 +59,7 @@ struct
     in
     cv false e
```

Comment:
I think we can remove the unused things here

Reply:
I removed the unused ones and completely split up this file to more analysis-specific ones because the remaining functions weren't necessarily for general-purpose use, but some analysis-specific logic, especially regarding a custom equality implementation.